common-after.js

```diff
@@ -608,32 +610,192 @@ function parseJSONData(data) {
       suddenly = false;
     }
   }
+  // Character card fields
+  if('PowerName' in data) {
+    $('#inputPowerName').val(data.PowerName);
+  } else {
+    $('#inputPowerName').val('');
+  }
+  if('NemesisIconURL' in data && data.NemesisIconURL.length != 0) {
+    loadedUserImages['nemesisIcon'] = new Image();
+    loadedUserImages['nemesisIcon'].src = data.NemesisIconURL;
+    loadedUserImages['nemesisIcon'].onload = function () {
+      // Once the Image has loaded, redraw the canvas so it immediately appears
+      drawCardCanvas();
```

Comment:
We're already calling `drawCardCanvas()` at the bottom of the function, and (I think) JS is just going to execute all of the logic here at once before it update the DOM, so I don't think we need any additional calls to `drawCardCanvas()` in places like here / the background art / etc.

Reply:
Hmmm, so if I remove all of these drawCardCanvas calls (except for the last one on line 735), once the JSON is parsed, the images are all still blank. I think the drawCardCanvas call is executed before the images are loaded, and so they don't get rendered properly. If I make any changes after the JSON is parsed, then the images render. 
Is there a way we can make a listener that waits for all the images to be loaded and then calls drawCardCanvas once that happens?

Reply:
Something about that seems off and I don't know what it is. We're running everything in order in our single thread, so calling the draw here vs above shouldn't be an issue... at least I think. Could you post the JSON you're using for your test so I can repro on my end?

One thing we could try is a deferred function to force the `drawCardCanvas()` call into a new thread:

```javascript
setTimeout(drawCardCanvas, 0);
```

My gut says that won't work, but it's probably worth a try.

Reply:
Here's an  example of a JSON that doesn't load immediately: https://gist.github.com/sjzhu/85698581df60ed4bb3fcc3aeae6d36ed
![image](https://github.com/Colcoction/unitys-workshop/assets/1555041/43c19597-6baa-4475-b940-f46b52a42c78)
After changing the HP value in the form field:
![image](https://github.com/Colcoction/unitys-workshop/assets/1555041/20d37951-6179-43ac-906c-ce5c339d1328)